crates/lib/src/qpu/api.rs

```diff
@@ -356,6 +326,55 @@ impl ExecutionOptions {
     }
 }
 
+#[cached(
+    result = true,
+    time = 60,
+    key = "String",
+    convert = r"{ String::from(quantum_processor_id)}"
+)]
+async fn get_accessor_with_cache(
+    quantum_processor_id: &str,
+    client: &Qcs,
+) -> Result<String, QpuApiError> {
+    println!("hydrating cache!");
+    get_accessor(quantum_processor_id, client).await
+}
+
+async fn get_accessor(quantum_processor_id: &str, client: &Qcs) -> Result<String, QpuApiError> {
+    let mut min = None;
+    let mut next_page_token = None;
+    loop {
+        let accessors = list_quantum_processor_accessors(
+            &client.get_openapi_client(),
+            quantum_processor_id,
+            Some(100),
+            next_page_token.as_deref(),
+        )
+        .await?;
+
+        let accessor = accessors
+            .accessors
+            .into_iter()
+            .filter(|acc| {
+                acc.live
+                // `as_deref` needed to work around the `Option<Box<_>>` type.
+                && acc.access_type.as_deref() == Some(&QuantumProcessorAccessorType::GatewayV1)
+            })
+            .min_by_key(|acc| acc.rank.unwrap_or(i64::MAX));
```

Comment:
Question for my understanding: how are these `rank` values set, and do we actually have situations where multiple accessors are returned from the API?

Reply:
They are set arbitrarily and that does not need to be known to the user. We do anticipate multiple values in a response

https://docs.api.qcs.rigetti.com/#operation/ListQuantumProcessorAccessors